lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -902,14 +982,22 @@ def _use_zordered_offset(self):
                 self._offsets = old_offset
 
     def _maybe_depth_shade_and_sort_colors(self, color_array):
-        color_array = (
-            _zalpha(color_array, self._vzs)
-            if self._vzs is not None and self._depthshade
-            else color_array
-        )
+        # Adjust the color_array alpha values if point depths are defined
+        # and depth shading is active
+        if self._vzs is not None and self._depthshade:
+            color_array = _zalpha(
+                color_array,
+                self._vzs,
+                min_alpha=self._depthshade_minalpha,
+                _data_scale=self._data_scale,
+            )
+
+        # Adjust the order of the color_array using the _z_markers_idx,
+        # which has been sorted by z-depth
         if len(color_array) > 1:
             color_array = color_array[self._z_markers_idx]
-        return mcolors.to_rgba_array(color_array, self._alpha)
+
+        return mcolors.to_rgba_array(color_array)
```

Comment:
Did we lose the alpha setting?

Reply:
In `_zalpha` we scale the existing alpha values by the depth shading "sats" multiplier. Is that what you're referring to?

```python
    # Change the alpha values of the colors using the generated alpha multipliers
    return np.column_stack([rgba[:, :3], rgba[:, 3] * sats])
```

Reply:
But that's not the same as `self._alpha`, is it? That's the alpha channel of the mapped colours.

Reply:
Ah, I see what you mean, good catch. Added back in